proxy.go

```diff
@@ -0,0 +1,343 @@
+package veneur
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"strconv"
+	"sync"
+	"syscall"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Sirupsen/logrus"
+	raven "github.com/getsentry/raven-go"
+	"github.com/hashicorp/consul/api"
+	"github.com/pkg/profile"
+	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/trace"
+	"github.com/zenazn/goji/bind"
+	"github.com/zenazn/goji/graceful"
+	"stathat.com/c/consistent"
+
+	"goji.io"
+	"goji.io/pat"
+)
+
+type Proxy struct {
+	Sentry                 *raven.Client
+	Hostname               string
+	ForwardDestinations    *consistent.Consistent
+	TraceDestinations      *consistent.Consistent
+	Discoverer             Discoverer
+	ConsulForwardService   string
+	ConsulTraceService     string
+	ConsulInterval         time.Duration
+	ForwardDestinationsMtx sync.Mutex
+	TraceDestinationsMtx   sync.Mutex
+	HTTPAddr               string
+	HTTPClient             *http.Client
+	Statsd                 *statsd.Client
```

Comment:
Did a quick run through the client. If unbuffered it just hits `Conn` which, [according to the docs](https://golang.org/pkg/net/#Conn), is fine. If buffered, it [uses a `sync.Mutex`](https://github.com/DataDog/datadog-go/blob/master/statsd/statsd.go#L166) to deal with multiple append calls. So in short, I think so!